apps/dashboard/src/app/(dashboard)/(chain)/[chain_id]/[contractAddress]/modules/components/CurrencySelector.tsx

```diff
@@ -0,0 +1,190 @@
+}
+
+export function CurrencySelector<
+  TFieldValues extends FieldValues = FieldValues,
+>({
+  small,
+  hideDefaultCurrencies,
+  showCustomCurrency = true,
+  isPaymentsSelector = false,
+  defaultCurrencies = [],
+  field,
+  contractChainId: chainId,
+}: CurrencySelectorProps<TFieldValues>) {
+  const { idToChain } = useAllChainsData();
+  const chain = chainId ? idToChain.get(chainId) : undefined;
+
+  const helperCurrencies =
+    defaultCurrencies.length > 0
+      ? defaultCurrencies
+      : chainId
+        ? CURRENCIES[chainId] || []
+        : [];
+
+  const [isAddingCurrency, setIsAddingCurrency] = useState(false);
+  const [editCustomCurrency, setEditCustomCurrency] = useState("");
+  const [customCurrency, setCustomCurrency] = useState("");
+  const [initialValue] = useState(field.value);
+
+  const isCustomCurrency: boolean = useMemo(() => {
+    if (initialValue && chainId && initialValue !== customCurrency) {
+      if (chainId in CURRENCIES) {
+        return !CURRENCIES[chainId]?.find(
+          (currency: CurrencyMetadata) =>
+            currency.address.toLowerCase() === initialValue.toLowerCase(),
+        );
+      }
+
+      // for non-default chains
+      return true;
+    }
+
+    return false;
+  }, [chainId, customCurrency, initialValue]);
+
+  const currencyOptions: CurrencyMetadata[] = [
+    ...(isPaymentsSelector
+      ? []
+      : [
+          {
+            address: NATIVE_TOKEN_ADDRESS.toLowerCase(),
+            name: chain?.nativeCurrency.name || "Native Token",
+            symbol: chain?.nativeCurrency.symbol || "",
+          },
+        ]),
+    ...(hideDefaultCurrencies ? [] : helperCurrencies),
+  ];
+
+  const addCustomCurrency = () => {
+    if (!isAddress(editCustomCurrency)) {
+      return;
+    }
+    if (editCustomCurrency) {
+      setCustomCurrency(editCustomCurrency);
+      if (field.onChange) {
+        field.onChange({
+          target: { value: editCustomCurrency },
+          // biome-ignore lint/suspicious/noExplicitAny: FIXME
+        } as any);
+      }
+    } else {
+      setEditCustomCurrency(customCurrency);
+    }
+
+    setIsAddingCurrency(false);
+    setEditCustomCurrency("");
+  };
+
+  if (isAddingCurrency && !hideDefaultCurrencies) {
+    return (
+      <div className="flex items-center">
+        <Button
+          className="rounded-r-none rounded-l-lg"
+          onClick={() => setIsAddingCurrency(false)}
+        >
+          &lt;-
+        </Button>
+        <Input
+          {...field}
+          className="w-full rounded-none"
+          required
+          placeholder="ERC20 Address"
+          value={editCustomCurrency}
+          onChange={(e) => setEditCustomCurrency(e.target.value)}
+        />
+        <Button
+          className="rounded-r-lg rounded-l-none"
+          onClick={addCustomCurrency}
+          disabled={!isAddress(editCustomCurrency)}
+        >
+          Save
+        </Button>
+      </div>
+    );
+  }
+
+  return (
+    <div
```

Comment:
remove this div wrapper and `small` prop and instead just expose a `className` prop and pass it to `Select` component

Reply:
Select doesn't have a className prop to it, so will leave the div wrapper, but introduced a className prop and removed the `small` prop

Reply:
Select is not a UI component - its a context component, so you need to pass the className to `SelectTrigger`